### PR TITLE
update tedlium3 recipe with transformer

### PR DESCRIPTION
**Pull request by ShigekiKarita**
# Transformer (elayers=12, dlayers=6, units=2048, 8 GPUs, specaug) + large LM

  - Model files (archived to tedlium3_largelm.tar.gz by `$ pack_model.sh`)
    - model link: (put the model link manually. please contact Shinji Watanabe <shinjiw@ieee.org> if you want a web storage to put your files)
    - training config file: `./conf/train_pytorch_transformer.v2.yaml`
    - decoding config file: `./conf/decode_pytorch_transformer_beam-size40.yaml`
    - cmvn file: `./data/train_trim_sp/cmvn.ark`
    - e2e file: `exp/train_trim_sp_pytorch_nbpe500_ngpu8_train_pytorch_transformer.v2_specaug/results/model.last10.avg.best`
    - e2e JSON file: `exp/train_trim_sp_pytorch_nbpe500_ngpu8_train_pytorch_transformer.v2_specaug/results/model.json`
    - lm file: `./exp/train_rnnlm_pytorch_lm_irie_unit1024_unigram500/rnnlm.model.best`
    - lm JSON file: `./exp/train_rnnlm_pytorch_lm_irie_unit1024_unigram500/model.json`
  - Results (paste them by yourself or obtained by `$ pack_model.sh --results <results>`)
```
exp/train_trim_sp_pytorch_nbpe500_ngpu8_train_pytorch_transformer.v2_specaug/decode_final_lmep8_beam40_dev_decode_pytorch_transformer_beam-size40/result.wrd.txt
|     SPKR                          |     # Snt          # Wrd     |     Corr             Sub            Del             Ins            Err           S.Err     |
|     Sum/Avg                       |      507           17783     |     91.7             4.7            3.5             1.5            9.7            74.6     |
exp/train_trim_sp_pytorch_nbpe500_ngpu8_train_pytorch_transformer.v2_specaug/decode_final_lmep8_beam40_test_decode_pytorch_transformer_beam-size40/result.wrd.txt
|     SPKR                       |     # Snt          # Wrd      |     Corr             Sub             Del             Ins             Err           S.Err      |
|     Sum/Avg                    |     1155           27500      |     92.8             3.6             3.6             0.8             8.0            67.4      |
```

## NOTE: contribution on WER

| system                     |   dev WER |   test WER |
| :------                    | --------: | ---------: |
| RNN                                   |     14.3 |        15.0 |  
| Transformer (large model, sp) |      13.4 |       10.0 |
| w/ specaug                 |      11.4 |        8.6 |
| w/ specaug + large LM      |       9.7 |        8.0 |

@sw005320 Could you put this packed model in Google Drive? https://send.firefox.com/download/167e0e0ad3be7bd0/#9wBFFsHlkv7oXnNl36Nq0Q

### CODE REVIEW

File: egs/tedlium3/asr1/conf/no_preprocess.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+process:
+  - type: "identity"
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/conf/no_preprocess.yaml]: This is not needed.
We just use `preprocss_conf=`.

File: egs/tedlium3/asr1/run.sh [hunk 2]
```diff
@@ -3,12 +3,12 @@
 # Copyright 2019 Nagoya University (Masao Someki)
 #  Apache 2.0  (http://www.apache.org/licenses/LICENSE-2.0)
 
-. ./path.sh || exit 1;
-. ./cmd.sh || exit 1;
+. ./path.sh
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/run.sh]:
```suggestion
. ./path.sh || exit 1;
```

File: egs/tedlium3/asr1/run.sh [hunk 3]
```diff
@@ -3,12 +3,12 @@
...
+. ./cmd.sh
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/run.sh]:
```suggestion
. ./cmd.sh || exit 1;
```

File: egs/tedlium3/asr1/run.sh [hunk 4]
```diff
@@ -81,8 +39,15 @@ bpemode=unigram
 # exp tag
 tag="" # tag for managing experiments.
 
+train_config=conf/train.yaml
+decode_config=conf/decode.yaml
+preprocess_config=conf/specaug.yaml
+
 . utils/parse_options.sh || exit 1;
 
+. ./path.sh
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/run.sh]:
```suggestion
```

File: egs/tedlium3/asr1/run.sh [hunk 5]
```diff
@@ -81,8 +39,15 @@ bpemode=unigram
...
+. ./cmd.sh
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/run.sh]:
```suggestion
```

File: egs/tedlium3/asr1/run.sh [hunk 6]
```diff
@@ -81,8 +39,15 @@ bpemode=unigram
...
+
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/run.sh]:
```suggestion
```

File: egs/tedlium3/asr1/run.sh [hunk 7]
```diff
@@ -81,8 +39,15 @@ bpemode=unigram
 # exp tag
 tag="" # tag for managing experiments.
 
+train_config=conf/train.yaml
+decode_config=conf/decode.yaml
+preprocess_config=conf/specaug.yaml
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/run.sh]: config files are defined twice.
Please remove the duplicated lines.

File: egs/tedlium3/asr1/run.sh [hunk 8]
```diff
@@ -232,22 +187,27 @@ if [ ${stage} -le 3 ] && [ ${stop_stage} -ge 3 ]; then
...
-        --patience ${lm_patience} \
-        --maxlen ${lm_maxlen} \
         --dict ${dict}
 fi
 
+if [ -z ${tag} ]; then
+    expname=${train_set}_${backend}_nbpe${nbpe}_ngpu${ngpu}_$(basename ${train_config%.*})_$(basename ${preprocess_config%.*})
+    if ${do_delta}; then
+        expname=${expname}_delta
+    fi
```
Comment (kan-bayashi) [inline egs/tedlium3/asr1/run.sh]: Could you add following part? 
https://github.com/espnet/espnet/blob/e8e43af565c4e0fb7b8b6f680c1ede485a5378f7/egs/librispeech/asr1/run.sh#L208-L210

File: egs/tedlium3/asr1/conf/no_preprocess.yaml (hunk 1 above)
Comment (kan-bayashi) [reply egs/tedlium3/asr1/conf/no_preprocess.yaml]: See `nargs=‘?'`
https://github.com/espnet/espnet/blob/e8e43af565c4e0fb7b8b6f680c1ede485a5378f7/espnet/bin/asr_train.py#L178

**Comment by ShigekiKarita** [comment]
done

**Comment by sw005320** [comment]
Many thanks!

**Pull request merged**